test/application.spec.ts

```diff
@@ -796,15 +800,19 @@ test.group('Application | requirePreloads', (group) => {
 			],
 		})
 
-		app.setup()
+		await app.setup()
 		app.registerProviders()
 		await app.bootProviders()
-		const fn = () => app.requirePreloads()
 
-		assert.throw(
-			fn,
-			`ENOENT: no such file or directory, open '${join(fs.basePath, 'start/foo.ts')}'`
-		)
+		try {
+			await app.requirePreloads()
+			assert.fail('unreachable')
+		} catch (err) {
+			assert.strictEqual(
+				err.message,
+				`ENOENT: no such file or directory, open '${join(fs.basePath, 'start/foo.ts')}'`
+			)
+		}
```

Comment:
is there a better way to verify that a promise rejects in Japa?

Reply:
Have to `try/catch`. But instead of using `assert.fail('unreachable')`. I will use assertion planning as follows:

```ts
assert.plan(1)

try {
  await app.requirePreloads()
} catch (err) {
  assert.strictEqual(
    err.message,
    `ENOENT: no such file or directory, open '${join(fs.basePath, 'start/foo.ts')}'`
  )
}
```

If the catch block is not executed, the test will still fail complaining that one assertion was planned and zero executed